File: Q-34.py
def noOfTest():
    T=input()
    T=int(T)
    return T

def noOfPairs(K):
    i=1
    j=i+1
    count=0
    while i<K-1:
        j=i+1
        while j<K:
            if i+j<=K:
                count=count+1
            j=j+1
        i=i+1
    return count

T=noOfTest()
if T>=1 and T<=100:
    K=list()

    for i in range(T):
        K.append(int(input()))

    for i in range(T):
        if K[i]>=1 and K[i]<=1000:
            print(noOfPairs(K[i]))

File: Q-34.c
#code
def printArray(A):
    for i in range(len(A)):
        print(A[i],end=" ")
    print()

t=int(input())
if t>=1 and t<=500:
    N=list()
    A=list()
    for i in range(t):
        N.append(int(input()))
        st=list(map(int,input().split()))
        A.append(st.sort())


    for i in range(t):
        if N[i]>=1 and N[i]<=10**6:
            printArray(A[i])
           
